tutorials/roofit/rf312_multirangefit.py

```diff
@@ -0,0 +1,106 @@
+#####################################
+#
+# 'MULTIDIMENSIONAL MODELS' ROOT.RooFit tutorial macro #312
+#
+# Performing fits in multiple (disjoint) ranges in one or more dimensions
+#
+#
+#
+# 07/2008 - Wouter Verkerke
+#
+# /
+
+
+import ROOT
+
+
+def rf312_multirangefit():
+
+    # C r e a t e   2 D   p d f   a n d   d a t a
+    # -------------------------------------------
+
+    # Define observables x,y
+    x = ROOT.RooRealVar("x", "x", -10, 10)
+    y = ROOT.RooRealVar("y", "y", -10, 10)
+
+    # Construct the signal pdf gauss(x)*gauss(y)
+    mx = ROOT.RooRealVar("mx", "mx", 1, -10, 10)
+    my = ROOT.RooRealVar("my", "my", 1, -10, 10)
+
+    gx = ROOT.RooGaussian("gx", "gx", x, mx, ROOT.RooFit.RooConst(1))
+    gy = ROOT.RooGaussian("gy", "gy", y, my, ROOT.RooFit.RooConst(1))
+
+    sig = ROOT.RooProdPdf("sig", "sig", gx, gy)
+
+    # Construct the background pdf (flat in x,y)
+    px = ROOT.RooPolynomial("px", "px", x)
+    py = ROOT.RooPolynomial("py", "py", y)
+    bkg = ROOT.RooProdPdf("bkg", "bkg", px, py)
+
+    # Construct the composite model sig+bkg
+    f = ROOT.RooRealVar("f", "f", 0., 1.)
+    model = ROOT.RooAddPdf("model", "model", ROOT.RooArgList(sig, bkg), ROOT.RooArgList(f))
+
+    # Sample 10000 events in (x,y) from the model
+    modelData = model.generate(ROOT.RooArgSet(x, y), 10000)
+
+    # D e f i n e   s i g n a l   a n d   s i d e b a n d   r e g i o n s
+    # -------------------------------------------------------------------
+
+    # Construct the SideBand1,SideBand2, regions
+    #
+    #                    |
+    #      +-------------+-----------+
```

Comment:
This comment might not look pretty in the notebook either. What we can do is just leave it for now, we see how the notebooks generated from your scripts look like, and then we decide what we can do.